### PR TITLE
Km 3851 input field spacing

### PR DESCRIPTION
**Pull request by kiley-mitti**
## Brief Description

I have replaced most all of the spacing css, and line height with relevant design tokens. Additionally I swapped border for shadow inset in order to create a consistent height while respecting design tokens.
During this process I discovered a number of additional spacing/sizing issues with icons within specific types of inputs. I've documented those more thoroughly along with possible suggestions for fixes in the Jira ticket below.

## JIRA Link

[ASTRO-3851](https://rocketcom.atlassian.net/browse/ASTRO-3851)

## Motivation and Context

Part of the Astro spacing initiative to move component css to design tokens where possible.

## Issues and Limitations

This ticket will be unable to be fully completed until we examine the icons of some of the input types and decide if we want to implement them differently.
Additionally I think we are missing one or two design tokens as they relate to -control- types.

## Types of changes

- [ ] Bug fix
- [ x] New feature
- [ ] Breaking change

## Checklist

- [ ] This PR adds or removes a Storybook story.
- [ ] I have added tests to cover my changes.
- [ ] Regressions are passing and/or failures are documented
- [x ] Changes have been checked in evergreen browsers

### CODE REVIEW

File: packages/web-components/src/components/rux-input/rux-input.scss [hunk 1]
```diff
@@ -62,57 +64,59 @@
...
         order: 2;
-        border: 1px solid var(--color-border-interactive-muted);
+        // use box shadow instead of border to create borders
+        box-shadow: var(--color-border-interactive-muted) 0 0 0 1px inset;
         border-radius: var(--radius-base);
         font-family: var(--font-body-1-font-family);
         font-size: var(--font-body-1-font-size);
         font-weight: var(--font-body-1-font-weight);
+        //control-body line heights not yet implemented
+        line-height: var(--line-height-sm);
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-input/rux-input.scss:74]: If you update `@astrouxds/design-tokens` to 2.0.0-beta.10, you should have access to `--font-control-body-1-line-height. 

nit: we should also update all the other references to `font-body-1` to `font-control-body-1`

File: packages/web-components/src/components/rux-input/rux-input.scss [hunk 2]
```diff
@@ -62,57 +64,59 @@
...
         letter-spacing: var(--font-body-1-letter-spacing);
         color: var(--color-text-primary);
         background-color: var(--color-background-base-default);
 
         &--invalid {
-            border: 1px solid var(--color-text-error);
+            box-shadow: var(--color-border-error) 0 0 0 1px inset;
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-input/rux-input.scss:80]: nice catch. we were using the wrong token (color-text-error)

File: packages/web-components/src/components/rux-input/rux-input.scss [hunk 3]
```diff
@@ -62,57 +64,59 @@
...
         &--large {
-            padding: 0.844rem 0.5rem; //13.5px 8px;
-            min-height: 2.875rem; //46px
+            padding: var(--spacing-3) var(--spacing-2); //12px 8px
         }
         &--search {
             -webkit-appearance: none;
             -moz-appearance: none;
-            padding-left: 2.25rem; //36px
+            padding-left: calc(var(--spacing-8)); //32px
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-input/rux-input.scss:99]: is this calc doing anything?

**Review by markacianfrani — CHANGES_REQUESTED** [review]
small note on missing line height token. 

in addition, can you also add a patch changeset please? reach out if ya need help with that

File: packages/web-components/src/components/rux-input/rux-input.scss (hunk 1 above)
Comment (kiley-mitti) [reply packages/web-components/src/components/rux-input/rux-input.scss]: Thank you! I didn't understand where the new ones were. I'll keep that in mind. Fixed. Also updated the other font-bodys to font-control-bodys

File: packages/web-components/src/components/rux-input/rux-input.scss (hunk 3 above)
Comment (kiley-mitti) [reply packages/web-components/src/components/rux-input/rux-input.scss]: Nope! It was but then I was able to remove the calculation and then my brain oozed out of my ears. Fixed!

**Review by markacianfrani — APPROVED** [review]
LGTM!

**Pull request merged**